ci/run.sh

```diff
@@ -36,17 +42,11 @@ case $1 in
         ;;
 esac
 
-NM=$(find $(rustc --print sysroot) -name llvm-nm)
+NM=$(find $(rustc --print sysroot) \( -name llvm-nm -o -name llvm-nm.exe \) )
```

Comment:
Seems like this is causing the issue: comparing a [working job](https://github.com/rust-lang/compiler-builtins/actions/runs/8038438058/job/21954371866#step:7:9389) to [the failing one](https://github.com/rust-lang/compiler-builtins/actions/runs/8412217195/job/23032821896?pr=582#step:7:6056) using `llvm-nm` instead of `nm` on i686-pc-windows-gnu is causing the tests to fail.

When I try to run `llvm-nm` locally, I get `error while loading shared libraries: ?: cannot open shared object file: No such file or directory` and an exit code of 127. Perhaps we're missing also missing some shared libraries required to run this on the CI machines?

Reply:
Could be libLLVM.dll itself. Try prefixing the command with `rustup run +nightly`. That will set the dynamic linker path as necessary.

Reply:
The test passes on windows, it's only macOS where it fails. See the CI results.

Reply:
> Could be libLLVM.dll itself. Try prefixing the command with `rustup run +nightly`. That will set the dynamic linker path as necessary.

Good call - that fixed the issue. I don't think it was libLLVM.dll (since that's not in the toolchain), but rather libgcc_s_dw2-1.dll and libwinpthread-1.dll.

I had to do a bit extra since not every build has `rustup` on the path, and the toolchain isn't always called `nightly`, but I now have this working: <https://github.com/rust-lang/compiler-builtins/pull/584/commits/424237207d6884aad7a0445496f0a8d2fdb96d08> @Amanieu feel free to steal that commit, or I can publish the draft PR if you prefer.

Reply:
Thanks for looking into this! I implemented a similar workaround based on yours but without the need for a second argument to run.sh.